src/main/scala/li/cil/oc/common/init/Items.scala

```diff
@@ -52,26 +52,26 @@ object Items extends ItemAPI {
     case _ => null
   }
 
-  def registerBlock[T <: Block](instance: T, id: String): T = {
+  def registerBlock(instance: Block, id: String): Block = {
     if (!descriptors.contains(id)) {
       instance match {
         case simple: SimpleBlock =>
           instance.setUnlocalizedName("oc." + id)
           instance.setRegistryName(id)
-          GameRegistry.register(instance)
+          GameData.register_impl(instance)
```

Comment:
in 1.12 you need to use the event! `register_impl` is an internal use method as far as I know. The models should also usually be registered in the event, after the block/item itself has registered